features/step_definitions/basic_steps.rb

```diff
@@ -10,21 +14,34 @@
   end
 end
 
+When(/^I fill in:$/) do |table|
+  table.hashes.each do |hash|
+      fill_in hash[:element], with: hash[:content]
+  end
+end
+
+When(/^I click the "([^"]*)" button$/) do |button|
+  click_button button
+end
+
+Then(/^I should see "([^"]*)"$/) do |content|
+  expect(page).to have_content content
+end
+
 When(/^I click the link "([^"]*)"$/) do |link|
   click_link(link)
 end
 
-Then(/^I should be on the "([^"]*)" page$/) do |page|
-  expected_page = goto(page)
-  expect(current_path).to eq expected_page
```

Comment:
But you didn't ask us not to have expectations on navigation steps. You asked us not to have them on given steps. With the test as it is now (corrected from the code above), we are expecting to be on a page after a button is clicked. We're testing a redirect. We'll eventually have another "visit a the "blah blah blah" page" where we will actually visit the page we expect to be on